src/main/java/edu/kis/vh/nursery/HanoiRhymer.java

```diff
@@ -1,18 +1,27 @@
-package edu.kis.vh.nursery;
-
-public class HanoiRhymer extends DefaultCountingOutRhymer {
-
-    int totalRejected = 0;
-
-    public int reportRejected() {
-        return totalRejected;
-    }
-
-    @Override
-    public void countIn(int in) {
-        if (!callCheck() && in > peekaboo())
-            totalRejected++;
-        else
-            super.countIn(in);
-    }
-}
+package edu.kis.vh.nursery;
+
+public final class HanoiRhymer extends DefaultCountingOutRhymer {
+
+    private static final int ZERO = 0;
+    private int totalRejected = ZERO;
+
+    int reportRejected() {
+        return getTotalRejected();
+    }
+
+    @Override
+    public void countIn(final int in) {
+        if (!callCheck() && in > peekaboo())
+            setTotalRejected(getTotalRejected() + 1);
+        else
+            super.countIn(in);
+    }
+
+    private int getTotalRejected() {
```

Comment:
Shouldn't be encapsulated (typical private field). This is an exaggeration. Using private fields internally usually do not require encapsulation.